src/lib/components/FormModel/children/Form/UI/ColorPicker/color-picker.component.js

```diff
@@ -1,47 +1,65 @@
-/* eslint-disable react/destructuring-assignment */
 import React, { useState } from 'react';
 import { ChromePicker } from 'react-color';
 
-const ColorPicker = ({ id, modifyFormObject, formObject, ...rest }) => {
-  const [show, setShow] = useState(false);
+import { FormModelUI } from '@constants';
+import { PickerGroup, ColorSwatch, Cover, Popover } from './color-picker.styles';
 
+const ColorPicker = ({
+  id,
+  modifyFormObject,
+  formObject,
+  value,
+  onSave,
+  autosave,
+  onBlur,
+  ...rest
+}) => {
+  let actualValue = formObject[id] || formObject[id] === '' ? formObject[id].value : value;
+
+  /* default value for the color */
+  if (!actualValue) actualValue = '#3498ef';
+
+  const [pickerVisible, setPickerVisible] = useState(false);
+  const [color, setColor] = useState(actualValue);
+
+  const label = rest[FormModelUI.UI_LABEL] || '';
   const handleChange = (color, event) => {
-    const obj = { value: color, ...rest };
+    event.preventDefault();
+    setColor(color.hex);
+  };
+
+  const handleChangeComplete = (color, event) => {
+    event.preventDefault();
+    const obj = { ...rest, value: color.hex };
     modifyFormObject(id, obj);
+    onSave();
```

Comment:
@james-martin-jd @Eliamar In this case I have to explicitly call onSave for the changes to be saved in the pod whether in other components calling `modifyFormObject` suffice. Can you comment why?